### DISCUSSION TITLE
migrate from 15.1.1 to  25.0.4

### DISCUSSION THREAD

**Discussion by adryen31** [keycloak/keycloak#33032] [Archive]
Hello all,

we are in keycloak 15.1.1 and we want on our Ubuntu server to upgrade to 25.0.4
do you know a link a or document, or process for upgrading from a old version to
this new version ? 
Do you have any idea to how to take care and what to do ?

Does I need to export our keycloak realm in command line from 15.1.1 to after
reimport it in keycloak 25.0.4 ? Or export database (postgres) directly to reimport
database after 25.0.4 launch ?
Should I keep a backup with any mean ?

Thank you so much and best regards

Adrien

**Comment by jonkoops** [comment]
Duplicate of #32743

**Reply by adryen31** [reply]
sorry but I need more informations. I can't directly install a new 25.0.4 version on the same server than the 15.1.1 with a same database connection ? (after make a backup of the database ?)
and use
bin/kc.sh start --http-relative-path /auth

?

Because is very complicated in an automated process to going to [21.1.2](https://github.com/keycloak/keycloak/releases/tag/21.1.2), [22.0.5](https://github.com/keycloak/keycloak/releases/tag/22.0.5), [23.0.7](https://github.com/keycloak/keycloak/releases/tag/23.0.7), [24.0.5](https://github.com/keycloak/keycloak/releases/tag/24.0.5) and finally [25.0.5](https://github.com/keycloak/keycloak/releases/tag/25.0.5)

**Reply by jonkoops** [reply]
Every upgrade will come with manual steps you will need take. Consult our [upgrading guide](https://www.keycloak.org/docs/latest/upgrading/index.html) for more information.
